src/main/scala/beam/agentsim/agents/PersonAgent.scala

```diff
@@ -763,8 +768,31 @@ class PersonAgent(
       )
   }
 
-  onTransition { case _ -> _ =>
-    unstashAll()
+  onTransition {
+    // todo rrp
+    case _ -> PlanningEnRouteCharging =>
+      nextStateData match {
+        case data: BasePersonData =>
+          val vehicleOpt = data.restOfCurrentTrip.headOption.map(leg => beamVehicles(leg.beamVehicleId).vehicle)
+          vehicleOpt match {
+            case None => log.error("PlanningEnRouteCharging: vehicle not found.")
+            case Some(vehicle) =>
+              chargingNetworkManager ! ParkingInquiry(
+                destinationUtm = vehicle.spaceTime,
+                activityType = ParkingActivityType.Charge,
+                vehicleManagerId = vehicle.vehicleManagerId,
+                beamVehicle = Some(vehicle),
+                valueOfTime = attributes.valueOfTime,
+                triggerId = getCurrentTriggerIdOrGenerate,
+                chargingPointTypes =
```

Comment:
@haitamlaarabi, I made the change as you have suggested. We will use `activityType: ParkingActivityType` of `ParkingInquiry` rather than adding new property `chargingPointTypes`. The "filter" logic is updated as well, so if activity type is `FastCharge`, look for >= 50 Kw power otherwise any will do (as long as we have charging point). There is a doubt, but I'll ask that once I push new changes.

Reply:
@haitamlaarabi, while skimming though charging point related code, I found we already have a threshold and predicate defined [here](https://github.com/LBNL-UCB-STI/beam/blob/rrp/calculate-battery-level-and-charging-point-stay-time/src/main/scala/beam/agentsim/infrastructure/charging/ChargingPointType.scala#L179) to check for "fast charger". Apparently, we use in `ifRideHailCurrentlyOnShiftThenFastChargingOnly` method, for the same purpose, to filter out charging stations. Should I make changes in existing thing or continue doing what I've done (you can check in newer commits)?

Reply:
Yes, feel free to use that one. You can also parameterized it so that we can change that 20kW from config file if we want to.

Reply:
You can also add a function similar to: `ifRideHailCurrentlyOnShiftThenFastChargingOnly`, or use the same by addition not only RideHail, but also private EVs who want to EnRoute charge. so instead of ParkingActivityType.FastCharge, create ParkingActivityType.EnRouteCharge